extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/JacksonMessageBodyReader.java

```diff
@@ -31,7 +31,11 @@ public JacksonMessageBodyReader(ObjectMapper mapper) {
     @Override
     public Object readFrom(Class<Object> type, Type genericType, Annotation[] annotations, MediaType mediaType,
             MultivaluedMap<String, String> httpHeaders, InputStream entityStream) throws IOException, WebApplicationException {
-        throw new IllegalStateException("Should never be called");
+        try {
+            return doReadFrom(type, genericType, entityStream);
+        } catch (MismatchedInputException e) {
+            throw new WebApplicationException(Response.Status.BAD_REQUEST);
```

Comment:
Looks like the original exception is lost here

Reply:
Yup, we don't send it back

Reply:
But shouldn't we at least `log.debug(..)` it?

Reply:
Yeah, I guess we can do that